FirebaseAuth/Sources/Swift/Utilities/AuthRecaptchaVerifier.swift

```diff
@@ -12,86 +12,71 @@
-    }
-  }
+@available(iOS 13, tvOS 13, macOS 10.15, macCatalyst 13, watchOS 7, *)
+class AuthRecaptchaConfig {
+  var siteKey: String?
+  let enablementStatus: [AuthRecaptchaProvider: AuthRecaptchaEnablementStatus]
 
-  @available(iOS 13, tvOS 13, macOS 10.15, macCatalyst 13, watchOS 7, *)
-  enum AuthRecaptchaEnablementStatus: String, CaseIterable {
-    case enforce = "ENFORCE"
-    case audit = "AUDIT"
-    case off = "OFF"
-
-    // Convenience property for mapping values
-    var stringValue: String { rawValue }
+  init(siteKey: String? = nil,
+       enablementStatus: [AuthRecaptchaProvider: AuthRecaptchaEnablementStatus]) {
+    self.siteKey = siteKey
+    self.enablementStatus = enablementStatus
   }
+}
 
-  @available(iOS 13, tvOS 13, macOS 10.15, macCatalyst 13, watchOS 7, *)
-  enum AuthRecaptchaProvider: String, CaseIterable {
-    case password = "EMAIL_PASSWORD_PROVIDER"
-    case phone = "PHONE_PROVIDER"
+@available(iOS 13, tvOS 13, macOS 10.15, macCatalyst 13, watchOS 7, *)
+enum AuthRecaptchaEnablementStatus: String, CaseIterable {
+  case enforce = "ENFORCE"
+  case audit = "AUDIT"
+  case off = "OFF"
 
-    // Convenience property for mapping values
-    var stringValue: String { rawValue }
-  }
+  // Convenience property for mapping values
+  var stringValue: String { rawValue }
+}
 
-  @available(iOS 13, tvOS 13, macOS 10.15, macCatalyst 13, watchOS 7, *)
-  enum AuthRecaptchaAction: String {
-    case defaultAction
-    case signInWithPassword
-    case getOobCode
-    case signUpPassword
-    case sendVerificationCode
-    case mfaSmsSignIn
-    case mfaSmsEnrollment
+@available(iOS 13, tvOS 13, macOS 10.15, macCatalyst 13, watchOS 7, *)
+enum AuthRecaptchaProvider: String, CaseIterable {
+  case password = "EMAIL_PASSWORD_PROVIDER"
+  case phone = "PHONE_PROVIDER"
 
-    // Convenience property for mapping values
-    var stringValue: String { rawValue }
-  }
+  // Convenience property for mapping values
+  var stringValue: String { rawValue }
+}
 
-  @available(iOS 13, tvOS 13, macOS 10.15, macCatalyst 13, watchOS 7, *)
-  class AuthRecaptchaVerifier {
-    private(set) weak var auth: Auth?
-    private(set) var agentConfig: AuthRecaptchaConfig?
-    private(set) var tenantConfigs: [String: AuthRecaptchaConfig] = [:]
-    private(set) var recaptchaClient: RCARecaptchaClientProtocol?
-    private static var _shared = AuthRecaptchaVerifier()
-    private let kRecaptchaVersion = "RECAPTCHA_ENTERPRISE"
-    init() {}
+@available(iOS 13, tvOS 13, macOS 10.15, macCatalyst 13, watchOS 7, *)
+enum AuthRecaptchaAction: String {
+  case defaultAction
+  case signInWithPassword
+  case getOobCode
+  case signUpPassword
+  case sendVerificationCode
+  case mfaSmsSignIn
+  case mfaSmsEnrollment
 
-    class func shared(auth: Auth?) -> AuthRecaptchaVerifier {
-      if _shared.auth != auth {
-        _shared.agentConfig = nil
-        _shared.tenantConfigs = [:]
-        _shared.auth = auth
-      }
-      return _shared
-    }
```

Comment:
With the sample app, I tried the following using the current code (red diff):
1. initialize recaptcha (success)
2. switch to configured secondary app
3. initialize recaptcha (threw internal error) 

I tried these steps again with the new code, and I got the same result.